tools/AduCmdlets/README.md

```diff
@@ -0,0 +1,209 @@
+
+2. Create or get a reference to an Azure Storage Blob container. `AduImportUpdate.psm1` module provides a helper cmdlet for this:
+
+    ```powershell
+    Import-Module .\AduImportUpdate.psm1
+
+    $AzureSubscriptionId = 'example'
+    $AzureResourceGroupName = 'example'
+    $AzureStorageAccountName = 'example'
+    $AzureBlobContainerName =  'example'
+
+    $container = Get-AduAzBlobContainer `
+        -SubscriptionId $AzureSubscriptionId `
+        -ResourceGroupName $AzureResourceGroupName `
+        -StorageAccountName $AzureStorageAccountName `
+        -ContainerName $AzureBlobContainerName
+    ```
+
+3. [Create a public Azure Active Directory (AzureAD) Client Application](https://docs.microsoft.com/en-us/azure/iot-hub-device-update/device-update-control-access#authenticate-to-device-update-rest-apis-for-publishing-and-management) if you do not already have one. This is required for calling ADU REST API.
+
+4. Obtain an OAuth authorization token for the client application. One option is to use [MSAL.PS](https://github.com/AzureAD/MSAL.PS) PowerShell module:
+
+    ```powershell
+    Install-Module MSAL.PS
+
+    $AzureAdClientId = 'example' # AzureAD application (client) ID.
+    $AzureAdTenantId = 'example' # AzureAD application directory (tenant) ID.
+
+    $token = Get-MsalToken `
+        -ClientId $AzureAdClientId `
+        -TenantId $AzureAdTenantId `
+        -Scopes 'https://api.adu.microsoft.com/user_impersonation' `
+        -Authority https://login.microsoftonline.com/$AzureAdTenantId/v2.0 `
+        -Interactive `
+        -DeviceCode
+    ```
+
+    **Note**: `MSAL.PS` cannot be loaded into the same session as `AzPowershell` due to an [unresolved issue](https://github.com/AzureAD/MSAL.PS/issues/32). For workaround, use them in separate PowerShell sessions.
+
+5. `ImportSampleBundleUpdate.ps1` script contains the sample code for creating a bundle update, and importing it to ADU using
+REST API. The update uses this README file as a payload and cannot actually be installed to a device.
+
+    ```powershell
+    $AduAccountEndpoint = 'example.api.adu.microsoft.com'
+    $AduInstanceId = 'example.instance'
+
+    .\ImportSampleBundleUpdate.ps1 `
+        -AccountEndpoint $AduAccountEndpoint `
+        -InstanceId $AduInstanceId `
+        -Container $container `
+        -AuthorizationToken $token `
+        -Verbose
+    ```
+
+### Additional Documentation
+
+To get detailed help information:
+
+```powershell
+Get-Help New-AduUpdateId -Detailed
+Get-Help New-AduUpdateCompatibility -Detailed
+Get-Help New-AduImportManifest -Detailed
+Get-Help Get-AduFileHashes -Detailed
+Get-Help New-AduImportUpdateInput -Detailed
+Get-Help Start-AduImportUpdate -Detailed
+Get-Help Wait-AduUpdateOperation -Detailed
+```
+
+## Bash Script
+
```

Comment:
i prefer two files.  If I'm using bash why should I wade through half of this document to find what I need?

Reply:
Added TOC.

GitHub would render README.md when user navigates to the directory. Having 2 readme files prevent that.